practices/software_portability.md

```diff
@@ -0,0 +1,89 @@
+---
+slug: software-portability
+stages:
+  - development
+short_description: Portability in high-level computer programming is the usability of the same software in different environments. 
+tags:
+  - software portability
+  - application
+  - source code
+  - data
+  - redevelopment
+keywords:
+  - software
+  - portability
+  - cross-platform
+  - redevelopment
+---
+
+# Software Portability
+
+## TL;DR
+
+Portability in high-level computer programming is the usability of the same software in different environments.
+
+## What Is Software Portability?
+
+Software portability is ability to move software from one machine platform to another. It refers to system software or application software that can be recompiled for a different platform or to software that is available for two or more different platforms. 
+
+## Why Should We Port?
+
+- There are many hardware and software platforms; it is not only the Windows world.
+- Users who move to different environments want familiar software.
+- We want easier migration to new system versions and to totally new environments.
+- Developers want to spend more time on new development and less on redevelopment.
+- More users for the same product means lower software costs.
+
+## Portability Kinds
+
+### Application Portability
+
+Portable applications can be converted from one computer environment to another one. 
+With regards to Windows apps, such programs would not use the Registry. In a totally portable application, all related files would be stored in a single folder so it can be copied with one command. Such a folder has many sub-folders, and they are copied in the same transaction. 
+[Walled gardens](https://www.pcmag.com/encyclopedia/term/walled-garden) prevent data from being easily transferred. 
+
+###  Source Code Portability
+
+Software can be compiled and linked from source code for different operating systems and processors if written in a programming language supporting compilation for the platforms. 
+This is usually a task for the program developers; typical users have neither access to the source code nor the required skills.
+
+### Data Portability 
+
+Portable data can be moved from one database or repository to another. It implies that the data are in electronic files that are readily accessible rather than only visually on a screen. Otherwise, the data must be retyped word for word into another application and on-screen images must be captured one at a time. 
+
+## Strategies for Portability
+
+-   Transferring installed program files to another computer of basically the same architecture.
+-   Reinstalling a program from distribution files on another computer of basically the same architecture.
+-   Building executable programs for different platforms from source code; this is usually called “porting”. 
+
+## Cross-platform Software 
+
+It is computer software that is implemented on multiple computing platforms. Cross-platform software may be divided into two types:
```

Comment:
(..) is computer software that is implemented on multiple computing platforms. Cross-platform software may be divided into two types; one requires individual building or compilation for each platform that it supports, and the other one can be directly run on any platform without special preparation.

For example, a cross-platform application may run on Microsoft Windows, Linux, and macOS.

Please, consider rewording this part.

Reply:
Wording changed.